prefq/server.py

```diff
@@ -55,14 +60,36 @@
 DEFAULT_DEBUG = False
 
 
-def before_first_request():
+def before_first_request(sshkey, server_pw):
     """Define starting routine"""
 
+    if sshkey is not None:
+        with open(sshkey, "rb") as key_file:
+            app.config["PRIVATE_KEY"] = serialization.load_ssh_private_key(
```

Comment:
Why does the server need a private key?

The way I would have imagined this key-based auth is:

- We use a challenge-response mechanism, i.e., the client first requests a challenge (a random string) from the server.
- The client uses its private key to encrypt the challenge and sends the response to the server.
- The server uses the client's public key to decrypt the response, comparing it with the challenge.
- If it matches, the client is authorized.

Alternatively we could consider just recommending an SSH tunnel in the README, providing less room for error.